### DISCUSSION TITLE
Split image to rectangles with no overlapping

### DISCUSSION THREAD

**Discussion by TarryPaloma** [ImageMagick/ImageMagick#1918] [General] [answered]
Hi,
Apologies as I'm a bit of a noob..
I'm trying to take an image of a bunch of rectangles, they may intersect or whatever, I want them split into as many individual rectangles with no overlaps as required.
I thought connected-components would do this, but the only way the output makes sense is if the objects overlap each other.
Attached input image and output "verbose" at the bottom of the post.
I would like to be able to re-construct the image from the output without having to overlap any shapes.

Is it even possible?

  1: 22x24+1+0 11.5,11.2 394 srgb(255,255,255)
  3: 10x24+14+0 19.2,12.6 60 srgb(29,255,72)
  0: 8x24+0+0 1.1,12.5 35 srgb(29,255,72)
  4: 6x9+4+7 6.5,10.8 34 srgb(255,242,29)
  7: 4x6+17+15 18.5,17.4 22 srgb(1,11,215)
  5: 4x3+5+9 6.5,10.0 12 srgb(255,255,255)
  2: 9x1+7+0 11.0,0.0 9 srgb(29,255,72)
  6: 4x2+5+13 6.5,13.5 8 srgb(255,255,255)
  8: 2x1+18+19 18.5,19.0 2 srgb(255,242,29)
![imagemagictomctest](https://user-images.githubusercontent.com/25393346/80029376-9b388480-84de-11ea-90b6-f6cc15877c93.png)

**Comment by fmw42** [comment, marked as answer]
What was your exact command line and what is your version of Imagemagick and platform?

**Reply by TarryPaloma** [reply]
Debian command prompt 7.0.10-7

magick convert ./images/imagemagictomctest.bmp -define connected-components:verbose=true  -define connected-components:area-threshold=1 connected-components:mean-color=true -connected-components 8 ./images/output.bmp

**Reply by fmw42** [reply]
Please explain what you want for the processed output image

**Reply by TarryPaloma** [reply]
I'm actually more interested in the verbose output, I want the dimensions and locations of the rectangles required to make up the image - but without having any overlapping. The idea being I could stick all the rectangles together in any order and they will re-make the source image.
I'll attempt to draw an explanation..

**Reply by TarryPaloma** [reply]
Drawn explanation..
I would want an output of these 35 discrete regions.
![imagemagic-whatamitryingtoachieve](https://user-images.githubusercontent.com/25393346/80032438-13a14480-84e3-11ea-8584-3c4ebe53e6f9.png)

**Reply by fmw42** [reply]
You cannot get those individually since many green section are connected. The software does not know that. It gives you the bounding boxes of each **connected** region.

**Reply by TarryPaloma** [reply]
ok, thanks for your help.

**Comment by fmw42** [comment]
-connected-components does not know about the contents of the bounding rectangles. So they will overlap. What it gives you is the bounds of each separate isolated regions. You have many green regions and you are getting the bounding boxes of each one. You would have to disconnect them.
